src/libPMacc/include/mpi/SeedPerRank.hpp

```diff
@@ -55,8 +56,18 @@ namespace mpi
         {
             PMACC_AUTO(&gc, PMacc::Environment<T_DIM>::get().GridController());
 
-            seed ^= gc.getGlobalSize( ) * localShift +
-                    gc.getGlobalRank( );
+            uint32_t rank = gc.getGlobalRank( );
+            /* We put the rank into the upper bits to allow values which start
+             * from zero (e.g. cellIdxs) to be used as additional seed contributors
+             * Those would then write to the lower bits leaving the upper bits alone
+             * which still results in globally unique seeds
+             */
+            uint32_t globalUniqueSeed = reverseBits(rank);
```

Comment:
I think to remember that we discuss this:

```C++
uint32_t rank = gc.getGlobalRank( );
rank ^= localShift; // still unique because it is only a rank mixing operation
unt32_t globalUniqueSeed = reverseBits(rank);
seed ^= globalUniqueSeed;
```